zingolib/src/wallet/transaction_records_by_id.rs

```diff
@@ -21,8 +31,94 @@ impl std::ops::DerefMut for TransactionRecordsById {
     }
 }
 impl TransactionRecordsById {
+    // initializers
+
     // Associated function to create a TransactionRecordMap from a HashMap
+    pub fn default() -> Self {
+        TransactionRecordsById(HashMap::new())
+    }
     pub fn from_map(map: HashMap<TxId, TransactionRecord>) -> Self {
         TransactionRecordsById(map)
     }
+
+    // modify methods
+
+    pub fn invalidate_all_transactions_after_or_at_height(&mut self, reorg_height: BlockHeight) {
+        // First, collect txids that need to be removed
+        let txids_to_remove = self
+            .values()
+            .filter_map(|transaction_metadata| {
+                if transaction_metadata
+                    .status
+                    .is_confirmed_after_or_at(&reorg_height)
```

Comment:
Should we have a simple wrapper that's `after_or_at` that combines broadcast and confirmed..  or is better to keep them more separate and explicit at this layer of abstractiong?